crates/hir_def/src/path.rs

```diff
@@ -167,8 +167,8 @@ pub enum GenericArg {
 impl Path {
     /// Converts an `ast::Path` to `Path`. Works with use trees.
     /// It correctly handles `$crate` based path from macro call.
-    pub fn from_src(path: ast::Path, hygiene: &Hygiene) -> Option<Path> {
-        lower::lower_path(path, hygiene)
+    pub fn from_src(path: ast::Path, ctx: &LowerCtx) -> Option<Path> {
```

Comment:
We need to pass the `LowerCtx` through, so further `TypeRef::from_ast` can access the `AstIdMap`.